pulsar-flink-connector/src/main/java/org/apache/flink/streaming/connectors/pulsar/internal/PulsarMetadataReader.java

```diff
@@ -357,17 +359,34 @@ public SchemaInfo getPulsarSchema(String topic) {
                 .collect(Collectors.toSet());
     }
 
+    /**
+     * Get topic partitions all, If the topic does not exist, it is created automatically ont partition to topic.
+     *
+     * @return allTopicPartitions
+     * @throws PulsarAdminException pulsarAdminException
+     */
     public Set<TopicRange> getTopicPartitionsAll() throws PulsarAdminException {
         List<TopicRange> topics = getTopics();
         HashSet<TopicRange> allTopics = new HashSet<>();
         for (TopicRange topic : topics) {
-            int partNum = admin.topics().getPartitionedTopicMetadata(topic.getTopic()).partitions;
+            int partNum = 1;
+            try {
+                partNum = admin.topics().getPartitionedTopicMetadata(topic.getTopic()).partitions;
+            } catch (PulsarAdminException.NotFoundException e) {
+                log.info("topic<{}> is not exit, auto create <{}> partition to <{}>", topic.getTopic(), partNum, topic.getTopic());
+                try {
+                    createTopic(topic.getTopic(), partNum);
+                } catch (PulsarAdminException.ConflictException conflictException) {
+                    // multi thread may cause concurrent creation
+                }
+            }
+            // pulsar still has the situation of getting 0 partitions, case: client.newProducer(Schema.STRING).topic(topicName).create()
             if (partNum == 0) {
```

Comment:
handle this problem.

https://github.com/streamnative/pulsar-flink/issues/366#issuecomment-873330849